tpm2/tpm2.go

```diff
@@ -0,0 +1,1247 @@
+	if err != nil {
+		return nil, err
+	}
+	b3, err := encodePasswordAuthArea(parentPassword, HandlePasswordSession)
+	if err != nil {
+		return nil, err
+	}
+	b4, err := tpmutil.Pack(toQuote, sigAlg)
+	if err != nil {
+		return nil, err
+	}
+	b5, err := encodeLongPCR(1, pcrNums)
+	if err != nil {
+		return nil, err
+	}
+	args := append(b1, b2...)
+	args = append(args, b3...)
+	args = append(args, b4...)
+	args = append(args, b5...)
+	return args, nil
+}
+
+func decodeQuote(in []byte) ([]byte, uint16, uint16, []byte, error) {
+	var empty []byte
+	var buf []byte
+	var attest []byte
+	var signature []byte
+	var s1 uint16
+	var s2 uint16
+
+	err := tpmutil.Unpack(in, &empty, &buf)
+	if err != nil {
+		return nil, 0, 0, nil, err
+	}
+
+	err = tpmutil.Unpack(buf, &attest, &s1, &s2, &signature)
+	if err != nil {
+		return nil, 0, 0, nil, err
+	}
+	return attest, s1, s2, signature, nil
+}
+
+// Quote returns a quote of PCR values.
+// Returns attestation data and the signature.
+func Quote(rw io.ReadWriter, signingHandle Handle, parentPassword, ownerPassword string, toQuote []byte, pcrNums []int, sigAlg Algorithm) ([]byte, []byte, error) {
+	cmd, err := encodeQuote(signingHandle, parentPassword, ownerPassword, toQuote, pcrNums, sigAlg)
+	if err != nil {
+		return nil, nil, err
+	}
+	resp, err := runCommand(rw, tagSessions, cmdQuote, tpmutil.RawBytes(cmd))
+	if err != nil {
+		return nil, nil, err
+	}
+	attest, _, _, sig, err := decodeQuote(resp)
+	if err != nil {
+		return nil, nil, fmt.Errorf("decoding Quote response: %v", err)
+	}
+	return attest, sig, nil
+}
+
+func encodeActivateCredential(activeHandle Handle, keyHandle Handle, activePassword, protectorPassword string, credBlob, secret []byte) ([]byte, error) {
+	b1, err := tpmutil.Pack(activeHandle)
+	if err != nil {
+		return nil, err
+	}
+	b2, err := tpmutil.Pack(keyHandle)
+	if err != nil {
+		return nil, err
+	}
+	b3, err := tpmutil.Pack([]byte(nil))
+	if err != nil {
+		return nil, err
+	}
+	b4a, err := encodePasswordAuthArea(activePassword, HandlePasswordSession)
+	if err != nil {
+		return nil, err
+	}
+	b4b, err := encodePasswordAuthArea(protectorPassword, HandlePasswordSession)
+	if err != nil {
+		return nil, err
+	}
+	b4t := append(b4a[2:], b4b[2:]...)
+	b4, err := tpmutil.Pack(b4t)
+	if err != nil {
+		return nil, err
+	}
+	b5, err := tpmutil.Pack(credBlob, secret)
+	if err != nil {
+		return nil, err
+	}
+	args := append(b1, b2...)
+	args = append(args, b3...)
+	args = append(args, b4...)
+	args = append(args, b5...)
+	return args, nil
+}
+
+func decodeActivateCredential(in []byte) ([]byte, error) {
+	var empty []byte
+	var buf []byte
+	var certInfo []byte
+
+	err := tpmutil.Unpack(in, &empty, &buf)
+	if err != nil {
+		return nil, err
+	}
+	err = tpmutil.Unpack(buf, &certInfo)
+	if err != nil {
+		return nil, err
+	}
+	return certInfo, nil
+}
+
+// ActivateCredential associates an object with a credential.
+// Returns decrypted certificate information.
```

Comment:
From just reading this comment, it's not clear what the certificate is that is getting decrypted. Please add a little more text here to make it clear to readers.